### PR TITLE
Parent object search fix

### PR DESCRIPTION
**Pull request by Tejass9922**
closes: #330 
---

## What
  * First draft of returning proper parents of the queried objects inside of the `RubiconJSON.search` function
  * There may potentially be some redundancy that needs to be looked over 

## How to Test
  *

### CODE REVIEW

File: rubicon_ml/client/rubicon_json.py [hunk 1]
```diff
@@ -214,6 +243,92 @@ def _rubicon_to_json(self, rubicon_objects):
 
         return json
 
+    def _fetch_parent_object(self, queried_object, id, return_type):
+        """
+        Returns the appropriate parent of the queried object to eventually append
+        to the return list of the query
+        """
+        if isinstance(queried_object, Project):
```
Comment (ryanSoley) [inline rubicon_ml/client/rubicon_json.py:251]: this will never be true, `match.value` won't ever be a rubicon object

**Comment by ryanSoley** [comment]
looks like the error in the tests is happening when there is a single project or single rubicon object provided to the class's init. the init should probably sanitize the input and make any single objects into lists

File: rubicon_ml/client/rubicon_json.py (hunk 1 above)
Comment (Tejass9922) [reply rubicon_ml/client/rubicon_json.py:251]: Good catch, thank you, I'll just use the return_type as an indicator

**Review by ryanSoley — COMMENTED** [review]
I'm trying to run this notebook from the jsonpath-poc integration branch (I think you need a rebases - doesn't look like the notebook is on your branch) and running into errors. looks like you're passing `return_type` to your `_fetch_query_string_type` and its only expecting `query_string`

File: rubicon_ml/client/rubicon_json.py [hunk 2]
```diff
@@ -76,79 +97,157 @@ def search(self, query, return_type=None):
...
+                            if key in match.value:
+                                del match.value[key]
+                    id = match.value["id"]
+                    parent = self._fetch_parent_object(
+                        queried_object=query_string_type, id=id, return_type="experiment"
+                    )
+                    return_objects.append(Experiment(DomainExperiment(**match.value), parent))
+
                             if key in match.value[i]:
                                 del match.value[i][key]
```
Comment (ryanSoley) [inline rubicon_ml/client/rubicon_json.py:157]: I think something got messed up in the merge here. I'm getting a syntax error

**Comment by ryanSoley** [comment]
closing for now in favor of #386 - we can revisit and complete this if the users ask for this additional functionality

**Pull request closed without merging**